### PR TITLE
Make Razor file encoding consistent

### LINKED ISSUE #11435: Blazor template files(.razor) contains inconsistent UTF-8 BOM
From https://github.com/aspnet/AspNetCore/issues/11225#issuecomment-504253296,

I created a new project in command-line with `dotnet new blazorserverside` and inspected each of the `.razor` file for BOM. Here is what I found,

Contains BOM:
- MyApp/Pages/_Imports.razor
- MyApp/Pages/Index.razor
- MyApp/Shared/MainLayout.razor
- MyApp/App.razor

Does not contain BOM:
- MyApp/_Imports.razor
- MyApp/Pages/_Host.cshtml
- MyApp/Pages/Counter.razor
- MyApp/Pages/FetchData.razor
- MyApp/Shared/NavMenu.razor

On top of this,
- Creating a `.razor` file from `Razor Component` item template creates a file with no BOM
- Creating a `.razor` file from `Razor View` or `Class` item template creates a file with BOM

I am not sure where the fix for this goes in to but we should make sure all our files are consistent. We should also make sure all of our other templates are consistent as well.

@danroth27 @ryanbrandenburg @mkArtakMSFT

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Fixes https://github.com/aspnet/AspNetCore/issues/11435

The diff here is basically the addition of UTF-8 BOM characters.

### CODE REVIEW

**Review by mkArtakMSFT — APPROVED** [review]
Thanks for adding the test!

File: src/ProjectTemplates/test/ByteOrderMarkTest.cs [hunk 1]
```diff
@@ -58,5 +58,32 @@ public void CheckForByteOrderMark_ForAllTemplates(string projectName)
...
+                foreach (var file in files)
+                {
+                    var filePath = Path.GetFullPath(file);
+                    var fileStream = new FileStream(filePath, FileMode.Open);
+
+                    var bytes = new byte[3];
+                    fileStream.Read(bytes, 0, 3);
+
+                    // Check for UTF8 BOM 0xEF,0xBB,0xBF
+                    Assert.True(bytes[0] == 0xEF && bytes[1] == 0xBB && bytes[2] == 0xBF, $"File {filePath} doesn't contains UTF-8 BOM characters.");
```
Comment (NTaylorMullen) [inline src/ProjectTemplates/test/ByteOrderMarkTest.cs]: I'd just compare with the mscorlib type `Utf8Encoding.GetPreamble()`

**Review by NTaylorMullen — APPROVED** [review]

File: src/ProjectTemplates/test/ByteOrderMarkTest.cs [hunk 2]
```diff
@@ -58,5 +58,32 @@ public void CheckForByteOrderMark_ForAllTemplates(string projectName)
 
             Assert.False(filesWithBOMCharactersPresent);
         }
+
+        [Fact]
+        public void RazorFilesInWebProjects_ShouldContainBOM()
```
Comment (ryanbrandenburg) [inline src/ProjectTemplates/test/ByteOrderMarkTest.cs]: Good test to have. Is there some context on why Razor should have BOM?

File: src/ProjectTemplates/test/ByteOrderMarkTest.cs [hunk 3]
```diff
@@ -58,5 +58,32 @@ public void CheckForByteOrderMark_ForAllTemplates(string projectName)
...
+            var directories = Directory.GetDirectories(Path.Combine(projectTemplateDir, path), "*Sharp");
+
+            foreach (var directory in directories)
+            {
+                var files = (IEnumerable<string>)Directory.GetFiles(directory, "*.cshtml", SearchOption.AllDirectories);
+                files = files.Concat(Directory.GetFiles(directory, "*.razor", SearchOption.AllDirectories));
+                foreach (var file in files)
+                {
+                    var filePath = Path.GetFullPath(file);
+                    var fileStream = new FileStream(filePath, FileMode.Open);
```
Comment (ryanbrandenburg) [inline src/ProjectTemplates/test/ByteOrderMarkTest.cs]: FileStream is `IDisposable`, this should be in a using.

**Review by ryanbrandenburg — APPROVED** [review]
One minor request, otherwise looks good.

File: src/ProjectTemplates/test/ByteOrderMarkTest.cs (hunk 2 above)
Comment (ajaybhargavb) [reply src/ProjectTemplates/test/ByteOrderMarkTest.cs]: > Is there some context on why Razor should have BOM?

Unfortunately I don't have any. @mkArtakMSFT and I spoke to the Web Tools team and they said they don't have any hard requirement on anything and VS respects the encoding of the original checked-in templates. My goal for this PR was to keep everything consistent once way or the other. I noticed most of the Razor files are with BOM and decided to make them all that way.

**Pull request merged**